### ISSUE TITLE
Update all actions to Node16 instead of Node12

### ISSUE THREAD

**Issue by shahzadlone** [sourcenetwork/defradb#979]
After #977 we only have to fix this remaining one:
- codecov/codecov-action@v2

TLDR: Update all actions from Node12 to Node16 by September 27th, 2023.

"
Node 12 has been out of support since [April 2022](https://github.com/nodejs/Release/#end-of-life-releases), as a result we have started the deprecation process of Node 12 for GitHub Actions. We plan to migrate all actions to run on Node16 by Summer 2023. We will monitor the progress of the migration and listen to the community for how things are going before we define a final date.
To raise awareness of the upcoming change, we are adding a warning into workflows which contain Actions running on Node 12. This will come into effect starting on September 27th.
"

Warning: "Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16"

Reference:
https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/
